.template/variants/api/app/controllers/concerns/localization.rb

```diff
@@ -2,12 +2,24 @@ module Localization
   extend ActiveSupport::Concern
 
   included do
-    before_action :set_locale
+    around_action :switch_locale
 
     protected
 
-    def set_locale
-      I18n.locale = request.headers[:'Accept-Language'] || I18n.default_locale
+    def switch_locale(&action)
+      locale = extract_locale_from_param || I18n.default_locale
```

Comment:
@andyduong1920 How about making a function which checks for both `params[:locale]` and `request.headers[:'Accept-Language']`
```
  def get_locale
    return params[:locale].to_sym if I18n.locale_available?(params[:locale])
    return request.headers[::'Accept-Language'].to_sym if I18n.locale_available?(request.headers[::'Accept-Language'].)

    I18n.default_locale
  end

```

Reply:
I think that depends on each project instead of the template @ankitkalia1195 some projects only allow param, some only allow header, some allow both. So the template should be something simple and general first, and depend on each project, they could customize it on demand.

Reply:
And also, this PR is for changing the `before_action` to `around_action` so for improving the `get locale` logic here, we can move to another issue/pr so we could discuss and implement on that, we also need to add Unit test for that too, those are out of this PR scope ;)

Reply:
@andyduong1920 Then why are we removing `request.headers[:'Accept-Language']`if we not changing the logic ?

Reply:
Oh yeah, the original version was `request.headers[:'Accept-Language']` 😁  so should we keep that or just keep going with param locale right now 🤔 

IMO we can go with that but not sure we have the different logic between API and Web variant.

Reply:
Changed in https://github.com/nimblehq/rails-templates/pull/241/commits/1cb5bc7b734584b070666284eae456d00584991a

Reply:
@andyduong1920 How about using both as I suggested and giving preference to one?